src/Types/Workout.ts

```diff
@@ -0,0 +1,114 @@
+import { Resources } from './Resources';
+export type WorkoutObjectType =
+  | 'exuipment'
+  | 'position'
+  | 'exercise'
+  | 'assignment'
+  | 'set'
+  | 'workout';
+
+export type Reference<T> = string | T;
+
+export interface WorkoutObject {
+  name: string; // unique string used to reference the object
+  label?: string; // string used for display, defaults to name
+  type?: WorkoutObjectType; // if not speified will be inffered
+  resources?: Reference<Resources>; // reference to a media resources object
+  // if not specified default media will be used
+}
+
+// e.g. palnk, down-facing-dog, transition
+// the name Position is not the best for a timed trnsition
+// Position duration can be left unspecified, or specified by trainer
+export interface Position extends WorkoutObject {
+  duration?: number; // Duration is oprional, may be assigned by trainer  // Referende to a positionobject
+}
+// Equipment metric can be left unspecified, or specified by trainer
+export interface Equipment extends WorkoutObject {
+  metric?: [number, number]; // range of weight, band number, etc
+  unit: string; // lbs for weight, band No. etc..
+}
```

Comment:
In my PR from last week, I suggested these changes to equipment to be more flexible with non-weighted equipment
```
export interface Equipment extends WorkoutObject {
  name: string; // Dumbbell, band, barbell, etc.
  equipmentUid: string;
  label?: string; // When resistance isn't a number. Red band, heavy weight, etc.
  resistance?: {
    range: [number, number]; // range of weight, band number, etc
    unit: string; // lbs for weight, band No. etc..
  };
}
```